databend-jdbc/src/main/java/com/databend/jdbc/DatabendConnection.java

```diff
@@ -106,6 +111,41 @@ public static String randRouteHint() {
         return sb.toString();
     }
 
+    private static final char SPECIAL_CHAR = '#';
+
+    public static String uriRouteHint(String URI) {
+        // Encode the URI using Base64
+        String encodedUri = Base64.getEncoder().encodeToString(URI.getBytes());
+
+        // Append the special character
+        return encodedUri + SPECIAL_CHAR;
```

Comment:
append a rand string after the encodedUri?

this may help make it a better load balance distribution at the chance of resetting the route hint on start query.

Reply:
we still expect using route hint with nginx to load balance the requests properly, so i think it's not bad to add some randomization here.

Reply:
oh ignore me, it's not make sense as route hint is expected to have such randomization